web/components/groups/group-banner-image.tsx

```
@@ -0,0 +1,259 @@
+  )
+  const [changeBannerModalOpen, setChangeBannerModalOpen] = useState(false)
+  return (
+    <>
+      <figure className="group relative h-60 w-full sm:h-72">
+        <div className="absolute top-2 right-4 z-20 transition-opacity sm:opacity-0 sm:group-hover:opacity-100">
+          {user && isEditable && (
+            <BannerDropdown
+              group={group}
+              open={changeBannerModalOpen}
+              setOpen={setChangeBannerModalOpen}
+              bannerUrl={groupBannerUrl}
+              setBannerUrl={setGroupBannerUrl}
+              user={user}
+              onChangeBannerClick={() => setChangeBannerModalOpen(true)}
+            />
+          )}
+        </div>
+        <Image src={groupBannerUrl} alt="" fill={true} objectFit="cover" />
+      </figure>
+    </>
+  )
+}
+
+async function updateGroupBannerImage(group: Group, bannerUrl?: string) {
+  if (!bannerUrl) {
+    return
+  }
+  if (group.bannerUrl && group.bannerUrl === bannerUrl) {
+    return
+  }
+  await updateGroup(group, { bannerUrl: bannerUrl })
+}
+
+export function ChangeBannerModal(props: {
+  group: Group
+  open: boolean
+  setOpen: (open: boolean) => void
+  bannerUrl: string
+  setBannerUrl: (bannerUrl: string) => void
+  user: User
+}) {
+  const { group, open, setOpen, bannerUrl, setBannerUrl, user } = props
+  const [bannerSelection, setBannerSelection] = useState(
+    isDefaultBanner(bannerUrl) ? bannerUrl : undefined
+  )
+  const [fileUrl, setFileUrl] = useState<string | undefined>(undefined)
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState(undefined)
+  const [fileLoading, setFileLoading] = useState(false)
+  const fileHandler = async (event: any) => {
+    const file = event.target.files[0]
+
+    setFileLoading(true)
+
+    await uploadImage(user.username, file)
+      .then(async (url) => {
+        setFileUrl(url)
+        setBannerSelection(url)
+        setFileLoading(false)
+      })
+      .catch(() => {
```

Did you mean `.finally(...)`? This would swallow the error.
Or, you could use `setError` here to inform the user.